saleor/graphql/tax/dataloaders.py

```diff
@@ -57,3 +60,22 @@ def batch_load(self, keys):
             keys
         )
         return [tax_class_map.get(obj_id) for obj_id in keys]
+
+
+class ProductChargeTaxesByTaxClassIdLoader(DataLoader):
+    context_key = "product_charge_taxes_by_tax_class_id"
+
+    def batch_load(self, keys):
+        non_zero_rates = TaxClassCountryRate.objects.filter(
+            tax_class=OuterRef("pk")
+        ).exclude(rate=Decimal(0))
+        tax_class_map = (
+            TaxClass.objects.filter(pk__in=keys)
+            .annotate(charge_taxes=Exists(non_zero_rates))
+            .in_bulk(keys)
+        )
+        return [
+            tax_class_map[tax_class_id].charge_taxes
+            for tax_class_id in keys
+            if tax_class_map.get(tax_class_id)
+        ]
```

Comment:
It can raise an error too, as AFAIK the output list from dataloader must be the same length as the input list.

Reply:
Hmm don't we have this problem in other dataloaders as well? I'm looking at usages of `in_bulk` in dataloaders and many seem to use similar pattern.

Reply:
The problem is in using `if` without an `else`. The data loader is a `map` function, and it needs to return the same number of results as there are keys and in the same order.

Reply:
Most other loaders use a `defaultdict` for the map.

Reply:
Thank you, fixed!